internal/envoy/envoy.go

```diff
@@ -213,6 +218,13 @@ func (srv *Server) run(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("error starting envoy: %w", err)
 	}
+	// call Wait to avoid zombie processes
+	go func() { _ = cmd.Wait() }()
```

Comment:
I'm not sure why `monitorProcessCancel` has to be inside `Server` - you can just wait for the process completion in this goroutine, and invoke `monitorProcessCancel` there.

Reply:
When we update envoy config we start a new envoy process using hot-reload. Once hot-reload completes it will kill the previous envoy process. If we don't cancel the previous process monitor, we will immediately exit when this happens. 
We have to cancel the process monitor **before** the process exits.